conjur-api/Resource.cs

```diff
@@ -110,5 +110,7 @@ public bool Check(string privilege)
                 offset += (uint)resultList.Count;
             } while (resultList.Count > 0);
         }
+
+        protected static string IdToName (string id, string account, ResourceKind kind) => id.Substring (id.IndexOf ($"{account}:{kind}:") + 1);
```

Comment:
Is there any reason not to just pass the id and return the last part? 
Also, this function would be clearer using full function definition syntax.

Reply:
Until now my impression is that colon is a valid symbol of resource name.
The next policy will be uploaded successfully
```
- !user
  id: abc:def
```
So simple splitting by colon and returning the last part will be definitely a bug.

Reply:
Well, you still don't need to pass account and kind explicitly. Here's how conjur itself does it: `self.id.split(':', 3)[2]`, cf.
https://github.com/cyberark/conjur/blob/37ddd2d51b6612edc52b430d0fd20174b501e08e/app/models/has_id.rb#L11

I believe C# equivalent would be almost the same, ie. `id.Split(":", 3)[2]`.

Reply:
Here I disagree with you.
Split does not make any sense here. Current implementation is self explained - resource id combined from account, kind and name. I think that it's important.
By the way colon is also valid symbol for kind (and may be for account also). I checked that the next policy works well for me:
```
- !resource &north_gate
  kind: ga:te
  id: north
```

Reply:
```
- !resource
  kind: ab:cd
  id: ef:gh
```
`conjur policy load root resource.yml`
Policy loaded successfully.
`curl -H "$(conjur authn authenticate -H)" http://conjur/resources/quick-start/ab%3Acd/ef%3Agh | jq .`
and
`curl -H "$(conjur authn authenticate -H)" http://conjur/resources/quick-start/ab/cd%3Aef%3Agh | jq .`
both return the right output
```
{
  "created_at": "2018-06-28T19:01:18.401+00:00",
  "id": "quick-start:ab:cd:ef:gh",
  "owner": "quick-start:user:admin",
  "policy": "quick-start:policy:root",
  "permissions": [],
  "annotations": []
}
```
So I still think that we need to use non-split parsing implementation.